src/screens/SendSheet.js

```diff
@@ -347,6 +304,87 @@ export default function SendSheet(props) {
     [selected, sendUpdateAssetAmount, updateMaxInputBalance]
   );
 
+  // Update all fields passed via params if needed
+  useEffect(() => {
+    if (recipientOverride && !recipient) {
+      setIsValidAddress(true);
+      setRecipient(recipientOverride);
+    }
+
+    if (assetOverride && assetOverride !== prevAssetOverride) {
+      sendUpdateSelected(assetOverride);
+      updateMaxInputBalance(assetOverride);
+    }
+
+    if (nativeAmountOverride && !amountDetails.assetAmount && maxInputBalance) {
+      sendUpdateAssetAmount(nativeAmountOverride);
+    }
+  }, [
+    amountDetails,
+    assetOverride,
+    maxInputBalance,
+    nativeAmountOverride,
+    prevAssetOverride,
+    recipient,
+    recipientOverride,
+    sendUpdateAssetAmount,
+    sendUpdateSelected,
+    updateMaxInputBalance,
+  ]);
+
+  const [isValidAddress, setIsValidAddress] = useState(!!recipientOverride);
+  const [currentProvider, setCurrentProvider] = useState();
+  const { colors, isDarkMode } = useTheme();
+
+  const showEmptyState = !isValidAddress;
+  const showAssetList = isValidAddress && isEmpty(selected);
+  const showAssetForm = isValidAddress && !isEmpty(selected);
```

Comment:
I've seen we use `useMemo` for stuff similar to this

Reply:
useMemo is a bit expensive to use too, so I try to use it whenever the logic it runs is worth memoizing. In this case isEmpty is not worth it.